### PR TITLE
Added EKS-to-EKS Clustermesh Preparation guide

### PR DESCRIPTION
**Pull request by network-charles**
Added EKS-to-EKS Clustermesh Preparation guide.

### CODE REVIEW

File: Documentation/network/clustermesh/eks-clustermesh-prep.rst [hunk 1]
```diff
@@ -0,0 +1,579 @@
...
+        apiVersion: eksctl.io/v1alpha5
+        kind: ClusterConfig
+
+        metadata:
+        name: ${NAME}
+        region: ${AWS_REGION}
+        vpc:
+          subnets:
+            private:
+              eu-west-2a:
```
Comment (giorio94) [inline Documentation/network/clustermesh/eks-clustermesh-prep.rst]:
```suggestion
              ${AWS_REGION}a: 
```

And the same for the one below.

File: Documentation/network/clustermesh/eks-clustermesh-prep.rst [hunk 2]
```diff
@@ -0,0 +1,579 @@
...
+            --protocol all \
+            --port 0 \
+            --source-group ${Cluster_2_SG}\
+            --region ${AWS_REGION}
+
+7. You now have a virtual private cloud, subnets, nat gateway, internet gateway, and a route table. You can create an EKS cluster without a CNI and request to use our custom VNet and subnet.
+
+    .. code:: bash
+
+        export NAME="$(whoami)-$RANDOM"
```
Comment (giorio94) [inline Documentation/network/clustermesh/eks-clustermesh-prep.rst]: Could be dropped, as already performed as first operation.

File: Documentation/network/clustermesh/eks-clustermesh-prep.rst [hunk 3]
```diff
@@ -0,0 +1,579 @@
...
+        apiVersion: eksctl.io/v1alpha5
+        kind: ClusterConfig
+
+        metadata:
+          name: ${NAME}
+          region: ${AWS_REGION}
+        vpc:
+          subnets:
+            private:
+              eu-west-2a:
```
Comment (giorio94) [inline Documentation/network/clustermesh/eks-clustermesh-prep.rst]:
```suggestion
              ${AWS_REGION}a: 
```

And the same for the one below.

File: Documentation/network/clustermesh/eks-clustermesh-prep.rst [hunk 4]
```diff
@@ -0,0 +1,579 @@
...
+            --protocol all \
+            --port 0 \
+            --source-group ${Cluster_1_SG}\
+            --region ${AWS_REGION}
+
+7. You now have a virtual private cloud, subnets, nat gateway, internet gateway, and a route table. You can create an EKS cluster without a CNI and request to use our custom VNet and subnet.
+
+    .. code:: bash
+
+        export NAME="$(whoami)-$RANDOM"
```
Comment (giorio94) [inline Documentation/network/clustermesh/eks-clustermesh-prep.rst]: Could be dropped, as already performed as first operation.

**Review by giorio94 — APPROVED** [review]
Thanks, looks great to me!

I've left just a couple of minor nits inline. I wonder if it could make sense to generalize the *Install cluster one/two* parts as they are pretty similar, to avoid the duplication and prevent future divergences. WDYT?

**Comment by giorio94** [comment]
I forgot to mention that there's a linting error: https://github.com/cilium/cilium/actions/runs/8952241106/job/24589411130?pr=32355

**Comment by network-charles** [comment]
@giorio94, I was attempting to squash some selected commits on the `eks-clustermesh-prep.rst` file but ended up squashing the ones in other files, so I had to close the PR, fix it and re-open the PR again.

I modified the content in `clustermesh.rst` file so it references `eks-clustermesh-prep.rst`, let me know what you think.

**Comment by giorio94** [comment]
>  I was attempting to squash some selected commits on the eks-clustermesh-prep.rst file but ended up squashing the ones in other files, so I had to close the PR, fix it and re-open the PR again.

No problem. Could you please squash the two remaining commits together? And remove my `Co-authored-by` line (it got probably added automatically when accepting a suggestion)?

> I modified the content in clustermesh.rst file so it references eks-clustermesh-prep.rst, let me know what you think.

Makes sense, thanks!

**Comment by network-charles** [comment]
> No problem. Could you please squash the two remaining commits together? And remove my `Co-authored-by` line (it got probably added automatically when accepting a suggestion)?

Done

**Comment by giorio94** [comment]
It seems that Anna is OOO. Would someone else from @cilium/docs-structure have time for a review?

File: Documentation/spelling_wordlist.txt [hunk 5]
```diff
@@ -4,6 +4,7 @@ Alemayhu
 Alibaba
 Aquantia
 Attestor
+AWS
```
Comment (ghost) [inline Documentation/spelling_wordlist.txt]: "aws" (lowercase) is already in the list. Lowercase entries should be case-insensitive, so uppercase "AWS" shouldn't be needed. Did the spell check fail without it?

**Review by ghost — COMMENTED** [review]
Sorry for delay! Looks good to me overall, just left one comment about the spelling wordlist.

File: Documentation/spelling_wordlist.txt (hunk 5 above)
Comment (network-charles) [reply Documentation/spelling_wordlist.txt]: No, it passed.
I have removed the uppercase "AWS".

**Review by ghost — APPROVED** [review]

**Pull request merged**